fe/fe-core/src/main/java/com/starrocks/analysis/ColumnDef.java

```diff
@@ -174,7 +174,7 @@ public void analyze(boolean isOlap) throws AnalysisException {
                         String.format("Cannot specify aggregate function '%s' for key column '%s'", aggregateType,
                                 name));
             }
-            if (!aggregateType.checkCompatibility(type.getPrimitiveType())) {
+            if (!(aggregateType.checkCompatibility(type.getPrimitiveType()) || (aggregateType.isReplaceFamily() && type.isArrayType()))) {
```

Comment:
`|| (aggregateType.isReplaceFamily() && type.isArrayType())`
IMO, it's better to place this check inside `ggregateType.checkCompatibility`

Reply:
the checkCompatibility is a simply table driven method, it check primitive types, the business logic of isArrayType, isHllType, isBitmapType, et al. is processed in analyze method right here.

Reply:
changed.